### PR TITLE
Correctly localize the "Failure" and "Warning" strings

### PR DESCRIPTION
**Pull request by jasonmalinowski**
*Review:* @dotnet/roslyn-ide 

Fixes part of internal bug 163139.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/WorkspaceDiagnostic.cs [hunk 1]
```diff
@@ -1,10 +1,11 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Diagnostics;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis
 {
-    [DebuggerDisplay("{GetDebuggerDisplay(),nq}")]
+    [DebuggerDisplay("{ToString(),nq}")]
```
Comment (davkean) [inline src/Workspaces/Core/Portable/Workspace/WorkspaceDiagnostic.cs]: This is now no longer needed, the debugger automatically calls ToString. You might be misremembering, when Roslyn expression evaluator broke this - but I've confirmed it's fixed in RTM.

**Comment by davkean** [comment]
:+1:

**Comment by davkean** [comment]
Re-syncing with Jenkins.

File: src/Workspaces/Core/Portable/Workspace/WorkspaceDiagnostic.cs [hunk 2]
```diff
@@ -18,13 +19,17 @@ public WorkspaceDiagnostic(WorkspaceDiagnosticKind kind, string message)
...
+            string kindText;
+
+            switch (Kind)
+            {
+                case WorkspaceDiagnosticKind.Failure: kindText = WorkspacesResources.Failure; break;
+                case WorkspaceDiagnosticKind.Warning: kindText = WorkspacesResources.Warning; break;
+                default: throw ExceptionUtilities.UnexpectedValue(Kind);
+            }
+
+            return string.Format("[{0}] {1}", kindText, this.Message);
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/Workspace/WorkspaceDiagnostic.cs]: I find your lack of string interpolation disappointing.

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**